File: app/src/main/java/com/example/pam_project/database/relationships/CategoriesWithLists.java
package com.example.pam_project.database.relationships;

import androidx.room.Embedded;
import androidx.room.Relation;

import com.example.pam_project.database.categories.CategoryEntity;
import com.example.pam_project.database.lists.ListEntity;

import java.util.List;

public class CategoriesWithLists {
    @Embedded
    public CategoryEntity category;
    @Relation(
            parentColumn = "id",
            entityColumn = "categoryId",
            entity = ListEntity.class
    )
    public List<ListsWithTasks> lists;
}

File: app/src/main/java/com/example/pam_project/database/relationships/CategoriesWithLists.kt
package com.example.pam_project.database.relationships

import androidx.room.Embedded
import androidx.room.Relation
import com.example.pam_project.database.categories.CategoryEntity
import com.example.pam_project.database.lists.ListEntity

data class CategoriesWithLists(@Embedded var category: CategoryEntity,
                               @Relation(parentColumn = "id", entityColumn = "categoryId", entity = ListEntity::class)
                               var lists: List<ListsWithTasks>)